app/news-page/components/edit-post-modal.tsx

```diff
@@ -1,48 +1,29 @@
-import { FormState } from 'formstate';
 import { observer } from 'mobx-react-lite';
 import * as React from 'react';
 
 import { Button, Form, Modal } from '@servicetitan/design-system';
 
 import { Label } from '../../common/components/label/label';
-import { InputFieldState, TextAreaFieldState } from '../../common/utils/form-helpers';
-import { FormValidators } from '../../common/utils/form-validators';
-import { Post } from '../models/post';
+import { EditPostStore } from '../stores/edit-post.store';
 
-function useStore() {
-    function factory() {
-        let post: Post | undefined;
-
-        return {
-            isOpen: false,
-            form: new FormState({
-                title: new InputFieldState(post && post.title).validators(
-                    FormValidators.required,
-                ),
-                description: new TextAreaFieldState(post && post.description).validators(
-                    FormValidators.required,
-                ),
-            }),
-            post,
-            onCancel: console.log,
-            onSave: console.log,
-        };
-    }
-
-    return React.useMemo(factory, []);
+interface EditPostModalProps {
+    store?: EditPostStore;
```

Comment:
It would be better to get the store instance from DI and have a component with `onCancel` and `onSave` methods.

Reply:
I don't see any benefits for static store instance. 
With static instance we need to set the post data when the modal opens, clear data after closing the modal and sync open/close state with NewsStore. Much easier to create a one-time store instance every time it needed.

Reply:
You could provide `EditPostStore` on the `EditPostModal` component then u wouldn't care about clean up and this component would be self-sufficient. But I've got your opinion, you could leave this as is if u wanna.